lib/components/result-view/display.js

```diff
@@ -1,27 +1,61 @@
 /* @flow */
 
 import React from "react";
+import { toJS } from "mobx";
 import {
   DisplayData,
   ExecuteResult,
   StreamText,
   KernelOutputError,
-  Output
+  Output,
+  Media,
+  RichMedia
 } from "@nteract/outputs";
-import { MediaComponents } from "./transforms";
+import PlotlyTransform from "@nteract/transform-plotly";
+import { VegaLite1, VegaLite2, Vega2, Vega3 } from "@nteract/transform-vega";
 
-type Props = {};
+type Props = { output: any };
 
-class Display extends React.PureComponent<Props> {
-  render() {
-    return (
-      <Output {...this.props}>
-        <ExecuteResult expanded>{MediaComponents()}</ExecuteResult>
-        <DisplayData expanded>{MediaComponents()}</DisplayData>
-        <StreamText expanded />
-        <KernelOutputError expanded />
-      </Output>
-    );
-  }
+// All supported media types for output go here
+export const supportedMediaTypes = (
+  <RichMedia>
+    <Vega3 />
+    <Vega2 />
+    <PlotlyTransform />
+    <VegaLite2 />
+    <VegaLite1 />
+    <Media.Json />
+    <Media.JavaScript />
+    <Media.HTML />
+    <Media.Markdown />
+    <Media.LaTeX />
+    <Media.SVG />
+    <Media.Image />
+    <Media.Plain />
+  </RichMedia>
+);
```

Comment:
I don't think that it is necessary to wrap them in a rich media, since react can interpret arrays as a group. The only place this would be necessary is if you tried to used this const by itself, but that could be seen as unclear inside a render function. If you desired use by itself, you could wrap the array with a RichMedia, and then it becomes much more clear whats going on without having to switch files to this declaration. It also technically uses a slight bit less memory with the array approach of course we are talking bytes here though.

Reply:
Yeah, this might seem a little weird right now since the idea of theis that this would all be one component (no array or instance like above).

The idea of the  `@nteract/outputs` api  is you just declare all the supported mediaTypes right in your component, in the order you want them prioritized, and the `Output` and `RichMedia` internals handle the rest. 

I don't think performance will come into play for this, it is more about keeping "transforms" in one place and _hopefully_ being easy to read. 

Since we have `ExecuteResult` and `DisplayData` we need to use this at least twice. I am about to open a PR on the monorepo to change that and get something like below:

```jsx
 function Display(props) {
  return (
    <Output output={toJS(props.output)}>
      <DisplayOrExecuteResult expanded>
        <Vega3 />
        <Vega2 />
        <PlotlyTransform />
        <VegaLite2 />
        <VegaLite1 />
        <Media.Json />
        <Media.JavaScript />
        <Media.HTML />
        <Media.Markdown />
        <Media.LaTeX />
        <Media.SVG />
        <Media.Image />
        <Media.Plain />
      </DisplayOrExecuteResult>
      <StreamText expanded />
      <KernelOutputError expanded />
    </Output>
  );
}

```